website/test_site_links.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+#
+# Quick script to help check for dead links in the markdown -> html converted
+# site that get's published to microsoft.github.io/MLOS
+#
+# Note: this assumes that build_site.sh, hugo, etc. have all already been run
+# to do the markdown -> html conversion.
```

Comment:
As in, if `./test_site_links.sh` is run prior to the site actually being generated?
Or, if some subdirectory that should have been generated wasn't quite right and `linklint` gives an obscure "such and such is missing" type of error that's then hard to track down because of the extra level of indirection?

Might be worth having a `README.md` about how the site is generated in the `website/` directory generally.

Reply:
yes, the site has actually been generated. I was just thinking about any hacky way to test this. But also seems like icing, was just an idea. A readme might be a good idea :)

Reply:
Added both.